lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.ts

```diff
@@ -136,9 +136,10 @@ export class AttachFileCloudWidgetComponent extends UploadCloudWidgetComponent i
             .subscribe((selections: Node[]) => {
                 selections.forEach(node => (node['isExternal'] = true));
                 const selectionWithoutDuplication = this.removeExistingSelection(selections);
+                const hadFilesAttached = this.field.value?.length > 0;
                 this.fixIncompatibilityFromPreviousAndNewForm(selectionWithoutDuplication);
-                if (this.field.value.length === 1) {
-                    this.contentModelFormFileHandler(selections && selections.length > 0 ? selections[0] : null);
+                if(!hadFilesAttached) {
+                    this.contentModelFormFileHandler(this.field.value.length > 0 ? this.field.value[0] : null);
                 }
```

Comment:
`this.fixIncompatibilityFromPreviousAndNewForm(selectionWithoutDuplication);` modifies `this.field.value`, so the if in line 140 was no working as expected

Reply:
Using `selections` was also not optimal - at this point `this.field.value` is modified - we can and should use it